qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -0,0 +1,764 @@
+                _check_keys("bounds")
+            else:
+                fitter_options["bounds"] = _dictionarize("bounds")
+        else:
+            fitter_options["bounds"] = dict(zip(fit_params, [(-np.inf, np.inf)] * len(fit_params)))
+
+        return fitter_options
+
+    def _subset_data(
+        self,
+        name: str,
+        series: np.ndarray,
+        x_values: np.ndarray,
+        y_values: np.ndarray,
+        y_sigmas: np.ndarray,
+    ) -> Tuple[np.ndarray, ...]:
+        """A helper method to extract reduced set of data.
+
+        Args:
+            name: Series name to search for.
+            series: An integer array representing a mapping of data location to series index.
+            x_values: Full data set of x values.
+            y_values: Full data set of y values.
+            y_sigmas: Full data set of y sigmas.
+
+        Returns:
+            Tuple of x values, y values, y sigmas for the specific series.
+
+        Raises:
+            AnalysisError:
+                - When name is not defined in the __series__ definition.
+        """
+        for idx, series_def in enumerate(self.__series__):
+            if series_def.name == name:
+                data_index = series == idx
+                return x_values[data_index], y_values[data_index], y_sigmas[data_index]
+        raise AnalysisError(f"Specified series {name} is not defined in this analysis.")
+
+    def _run_analysis(
+        self, experiment_data: ExperimentData, **options
+    ) -> Tuple[List[AnalysisResult], List["pyplot.Figure"]]:
+        """Run analysis on circuit data.
+
+        Args:
+            experiment_data: the experiment data to analyze.
+            options: kwarg options for analysis function.
+
+        Returns:
+            tuple: A pair ``(analysis_results, figures)`` where
+                   ``analysis_results`` may be a single or list of
+                   AnalysisResult objects, and ``figures`` is a list of any
+                   figures for the experiment.
+
+        Raises:
+            AnalysisError: if the analysis fails.
+        """
+        analysis_result = CurveAnalysisResult()
+        figures = list()
+
+        # pop arguments that are not given to fitter
+        curve_fitter = options.pop("curve_fitter")
+        data_processor = options.pop("data_processor")
+        x_key = options.pop("x_key")
+        plot = options.pop("plot")
+        axis = options.pop("ax")
+        xlabel = options.pop("xlabel")
+        ylabel = options.pop("ylabel")
+        fit_reports = options.pop("fit_reports")
+        return_data_points = options.pop("return_data_points")
+
+        #
+        # 1. Setup data processor
+        #
+        if isinstance(data_processor, DataProcessor) and not data_processor.is_trained:
+            # Qiskit DataProcessor instance. May need calibration.
+            try:
+                data_processor.train(data=experiment_data.data())
```

Comment:
Can we add a mechanism to train on a subset of the data which would default to training on all the data?

Reply:
I think this filtering can be offloaded to processor itself.